src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -19,6 +20,59 @@ internal partial class SQLitePersistentStorageService : AbstractPersistentStorag
 
         private readonly IPersistentStorageFaultInjector _faultInjectorOpt;
 
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr LoadLibrary(string dllToLoad);
+        private static bool _initialized;
+        private static object _gate = new object();
+
+        internal static bool TryInitializeLibraries()
+        {
+            if (_initialized)
+            {
+                return true;
+            }
+                
+            lock (_gate)
+            {
+                if (_initialized)
+                {
+                    return true;
+                }
+
+                // Attempt to load the correct version of e_sqlite.dll.  That way when we call
+                // into SQLitePCL.Batteries_V2.Init it will be able to find it.
+                //
+                // Only do this on Windows when we can safely do the LoadLibrary call to this
+                // direct dll.  On other platforms, it is the responsibility of the host to ensure
+                // that the necessary sqlite library has already been loaded such that SQLitePCL.Batteries_V2
+                // will be able to call into it.
+                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+                {
+                    var myFolder = Path.GetDirectoryName(
+                        typeof(SQLitePersistentStorage).Assembly.Location);
+
+                    var is64 = IntPtr.Size == 8;
+                    var subfolder = is64 ? "x64" : "x86";
+
+                    LoadLibrary(Path.Combine(myFolder, subfolder, "e_sqlite3.dll"));
+                }
+
+                try
+                {
+                    // Necessary to initialize SQLitePCL.
+                    SQLitePCL.Batteries_V2.Init();
+                }
+                catch (Exception e) when (e is DllNotFoundException || e is EntryPointNotFoundException)
+                {
+                    StorageDatabaseLogger.LogException(e);
+                    return false;
```

Comment:
i really dislike that we haven't marked _iniitialized here.  so this can just be hit over and over again.

Reply:
This was just to test whether unit tests passed with this. I used to have a `Lazy<bool>` in an earlier version, but that broke unit tests in the same way they are broken now.